framework/include/base/MooseObject.h

```diff
@@ -18,6 +18,7 @@
 // MOOSE includes
 #include "InputParameters.h"
 #include "ConsoleStreamInterface.h"
+#include "Parser.h"
```

Comment:
Oh boy... The Parser makes inroads further into MOOSE! This is as core as it gets.

Reply:
This actually doesn't have to go here.  We could move the special info param name functions (paramLocName, paramPathName) into MooseObject.h/C which Parser.C already has access to.